### PR TITLE
Tooling refinements

### PR DESCRIPTION
**Pull request by svandragt**
### Description
Tooling refinements and fixes split from #28. 

- Raises the default minimum environment target to PHP 5.6. 
- Move PHPCS / PHPUnit configuration into the xml configuration files (no effective change).
- Improvement in detection of installed code standards through added composer requirement.
- Simplified test pattern matching (no effective change).
- Remove behat configuration remainders.

Added a composer package to correctly detect all installed code sniffer standards. 

### Type of change
Please select the relevant options:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Steps to test
     composer install
     composer local:tests

### Acceptance criteris
- [x] My code follows WordPress coding standards.
- [x] I have performed a self-review of my own code.
- [x] If the changes are visual, I have cross browser / device tested.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added test instructions that prove my fix is effective or that my feature works.

### CODE REVIEW

**Comment by svandragt** [comment]
The test:phpcs command is effectively unchanged compared to before this commit, all I've done is move all the parameters inside the phpcs.xml.dist file, which is the file called by phpcs when none is specified.

I simplified the configuration based on the best practices suggested in https://dev.hmn.md/2019/10/10/making-the-best-of-config/.

This can be verified by changing the line  to `which phpcs` it displays _[...]plugins/core-sitemaps/vendor/bin/phpcs_  as for composer this is an equivalent command.

**Review by kirstyburgoine — APPROVED** [review]
👍

File: composer.json [hunk 1]
```diff
@@ -56,8 +56,9 @@
     "oomphinc/composer-installers-extender": "^1.1"
   },
   "require-dev": {
+    "dealerdirect/phpcodesniffer-composer-installer": "^0.5.0",
     "phpcompatibility/phpcompatibility-wp": "^2.1",
-    "phpunit/phpunit": "<6.0",
+    "phpunit/phpunit": "<8.0",
```
Comment (joemcgill) [inline composer.json]: Will have to check, but we might need to use the same phpunit version that WP Core uses. This is ok for now.

File: composer.json [hunk 2]
```diff
@@ -38,10 +38,10 @@
       "vagrant ssh -c \"cd /vagrant/content/plugins/core-sitemaps && WP_TESTS_DB_PASS=password composer run test:phpunit\""
     ],
     "test:phpcs": [
-      "vendor/bin/phpcs -nps --colors --report-code --report-summary --report-width=80 ."
+      "phpcs"
```
Comment (joemcgill) [inline composer.json:41]: I'd prefer to use the dependency version to ensure everyone is getting the same output. Should be: `vendor/squizlabs/php_codesniffer/bin/phpcs`.

**Review by joemcgill — APPROVED** [review]
Looking good. Update the location of phpcs and I think this is ready to go in. Going to pre-approve.

File: composer.json (hunk 2 above)
Comment (joemcgill) [reply composer.json:41]: FYI: for me, running `which phpcs` returns the globally installed path in my users directory.

Comment (swissspidy) [reply composer.json:41]: This should not be needed (i.e. the change is correct).

https://getcomposer.org/doc/articles/scripts.md#writing-custom-commands

File: composer.json (hunk 1 above)
Comment (svandragt) [reply composer.json]: I'm told core is going to use: `"phpunit/phpunit": "^5.0 || ^6.0 || ^7.0"` so i'll go with that then.

File: composer.json (hunk 2 above)
Comment (svandragt) [reply composer.json:41]: > FYI: for me, running `which phpcs` returns the globally installed path in my users directory.

You're probably running it in the terminal @joemcgill , whereas mean to say if that line is changed in the composer.json; ie. line-41: "which phpcs" composer then resolves to `vendor/bin/phpcs`.

Comment (joemcgill) [reply composer.json:41]: Ah yeah, misunderstood what you were suggesting. Wasn't aware that composer would prioritize a vendor version over whatever was globally available. Nice.

**Pull request merged**